### PR TITLE
#1115 create blog index page UI

### LINKED ISSUE #1115: Create blog index page
# Feature request
We need a page in `/pages/blog/index.js` that acts as a system of listing the available blogs and rendering that list as pretty links that display meta data.

## Proposed solution
The page should use SSG like the [dynamic blog routes file](https://github.com/OperationCode/front-end/blob/main/pages/blog/%5Barticle%5D.js) to get a list of the articles and pass the array as a prop. For now, let's just use lodash string functions to convert the filename to a human-readable title and list the articles as links.

[Documentation for `getStaticPaths`](https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation)

Let's emulate FreeCodeCamp's [News page](https://www.freecodecamp.org/news/) as inspiration for the design.

### PR DESCRIPTION
**Pull request by calincionca35**
# Description of changes

Creates Blog page UI using OC's ImageCard component
# Issue Resolved

Fixes #1115

### CODE REVIEW

**Comment by kylemh** [comment]
#### 📝Changed routes:


| [**/blog**](https://storybook-9kf1gzrq4-operation-code.vercel.app/blog) |
|:-:|
|<a href="https://storybook-9kf1gzrq4-operation-code.vercel.app/blog">Screenshot of /blog</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/ywzzWH0dnTnk3K0jYiju_pidCeA_.png">(view full size)</a> |







Commit f013b066996ce27b0fc72b85db43e949afc6141b (https://storybook-9kf1gzrq4-operation-code.vercel.app).

File: pages/blog/index.js [hunk 1]
```diff
@@ -29,14 +33,20 @@ function BlogIndex({ articles }) {
...
-            <li key={article}>
-              <Link href="/blog/[article]" as={`/blog/${article}`}>
-                <a>{article}</a>
-              </Link>
-            </li>
+            <ImageCard
+              alt="Two women pair programming"
+              imageSource={`${s3}stock_paired-programming.jpg`}
+            >
+              <li key={article}>
```
Comment (kylemh) [inline pages/blog/index.js:44]: Instead, I'd wrap the `ImagCard` inside `li`.

File: styles/blog.module.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.blogsList {
+  list-style-type: none;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 0;
+  margin: 0;
+}
```
Comment (kylemh) [inline styles/blog.module.css:8]: Would love to see if you could do 3 cards per row with flexbox and max-width on the container, and then it responsively adjusts to 2 and then 1 column.

File: pages/blog/index.js (hunk 1 above)
Comment (calincionca35) [reply pages/blog/index.js:44]: I will make this change.

File: styles/blog.module.css (hunk 2 above)
Comment (calincionca35) [reply styles/blog.module.css:8]: I think I can do this.

File: pages/blog/index.js (hunk 1 above)
Comment (calincionca35) [reply pages/blog/index.js:44]: Thank you for reviewing my PR. I will make this change.

File: styles/blog.module.css [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  list-style-type: none;
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-around;
+  padding: 0;
+  margin: 0;
+}
+
+.blog {
+  width: 33%;
```
Comment (kylemh) [inline styles/blog.module.css:11]: get rid of this

File: styles/blog.module.css [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  .blog {
+    width: 50%;
+  }
+}
+
+@media screen and (--small-viewport) {
+  .blog {
+    width: 100%;
+  }
+}
```
Comment (kylemh) [inline styles/blog.module.css:31]: and this

and you're done!

Comment (calincionca35) [reply styles/blog.module.css]: Thank you for the review.

**Review by kylemh — APPROVED** [review]

**Pull request merged**